### ISSUE TITLE
Unexpected `self.` autocorrection

### ISSUE THREAD

**Issue by anton-plebanovich** [nicklockwood/SwiftFormat#1112]
**Version:** `0.49.1` both `master` and `develop`

I have a code like:
```
extension CountryCodePickerViewModel: TypeAheadDelegate {
    
    func onSelect(viewController: TypeAheadViewController, textField: UITextField?, item: TypeAheadViewDataSource?) {
        if let countryCode = item as? CountryCodeModel,
           let selectedIndex = countryCodes.firstIndex(of: countryCode) {
            self.selectedIndex = selectedIndex
        }
    }
    
}
```

It corrects it to `selectedIndex = selectedIndex` which then produces an error `Cannot assign to value: 'selectedIndex' is a 'let' constant`.

**Comment by stevelandeyasana** [comment]
I'm also seeing this error as well, but weirdly just in one file. It seems this might be an issue inside conditionals.

**Comment by nicklockwood** [comment]
@anton-plebanovich @stevelandeyasana fixed in 0.49.2
